Core/include/Acts/Utilities/Intersection.hpp

```diff
@@ -256,29 +260,25 @@ class ObjectMultiIntersection {
 
 namespace detail {
 
-/// This function checks if an intersection is valid for the specified
-/// path-limit and overstep-limit
-///
-/// @tparam intersection_t Type of the intersection object
+/// This function checks if an intersection path length is valid for the
+/// specified near-limit and far-limit
 ///
-/// @param intersection The intersection to check
-/// @param nearLimit The minimum distance for an intersection to be considered
-/// @param farLimit The maximum distance for an intersection to be considered
+/// @param pathLength The path length of the intersection
+/// @param nearLimit The minimum path length for an intersection to be considered
+/// @param farLimit The maximum path length for an intersection to be considered
 /// @param logger A optionally supplied logger which prints out a lot of infos
 ///               at VERBOSE level
-template <typename intersection_t>
-bool checkIntersection(const intersection_t& intersection, double nearLimit,
-                       double farLimit,
-                       const Logger& logger = getDummyLogger()) {
-  const double distance = intersection.pathLength();
+inline bool checkPathLength(double pathLength, double nearLimit,
```

Comment:
we could compile this into a .cpp file, not sure if performance could suffer from this though...